folium/map.py

```diff
@@ -465,3 +465,35 @@ def __init__(self, bounds, padding_top_left=None,
         self.fit_bounds_options = json.dumps({key: val for key, val in
                                               options.items() if val},
                                              sort_keys=True)
+
+
+class CustomPane(MacroElement):
+    """
+    Creates a custom pane to hold map elements.
+
+    """
+    _template = Template(u"""
+        {% macro script(this, kwargs) %}
+            {{this._parent.get_name()}}.createPane('{{this.get_name()}}');
+            {{this._parent.get_name()}}.getPane(
+                '{{this.get_name()}}').style.zIndex = {{this.z_index}};
+            {% if not this.pointer_events %}
+                {{this._parent.get_name()}}.getPane(
+                    '{{this.get_name()}}').style.pointerEvents = 'none';
+            {% endif %}
+        {% endmacro %}
+        """)
+
+    def __init__(self, z_index=625, pointer_events=False):
+        super(CustomPane, self).__init__()
+        self._name = 'Pane'
+        self.z_index = str(z_index)
+        self.pointer_events = pointer_events
+
+    def render(self, **kwargs):
+        for name, element in self._children.items():
+            options = json.loads(element.options)
+            options.update({'pane': self.get_name()})
+            element.options = json.dumps(options, sort_keys=True, indent=8)
+            element._parent = self._parent
```

Comment:
When you move the element up the tree it will not get rendered, because the parent of `CustomPane` has already been rendered...

Reply:
It rendered OK in my test notebook, see screenshot below. I haven't properly moved the element up the tree -- just swapped out the parent, but the tile layer still exists in the pane's children list.

![image](https://user-images.githubusercontent.com/29736082/54478693-03616500-480d-11e9-9279-1dceef9524b5.png)

Reply:
Ah right, it's in the list of children so it will get rendered. Good one.

Reply:
Changed the parent just so that, when render is called on the layer, the javascript ends up as
```js
element().add_to(map_name)
```
rather than
```js
element().add_to(pane_name)
```
which leaflet wouldn't understand

Reply:
That makes sense, thanks for explaining.